src/sympc/tensor/replicatedshare_tensor.py

```diff
@@ -283,6 +283,8 @@ def sanity_checks(
             ValueError: if both RSTensor have different ring_sizes
         """
         if not isinstance(y, ReplicatedSharedTensor):
+            # For rsub in MPCTensor which multiplies it with (-1)
+            y = y % PRIME_NUMBER if x.ring_size == PRIME_NUMBER else y
```

Comment:
Q: Shouldn't we move the modulo there? (in rsub?)

Reply:
Yeah, that would be modular.I would move it. :+1:

Reply:
I retrospected,George It applies for rsub and sub and mul ops, Shall we keep it here itself?

Reply:
Maybe I will modify the comments regarding the same.

Reply:
Done , I have modified the comments!